### ISSUE TITLE
Handling help queries

### ISSUE THREAD

**Issue by salvipeter** [dennisog/julia-shell-mode#4]
In the standard Julia shell, pressing `?` at an empty prompt changes the prompt to `help?>`, and entering a word followed by a newline gives help on that word.

This is a very useful tool, and should be implemented in julia-shell-mode, as well. Note that calling `@doc` on the word is not really the same, because
- `@doc` does not give hints on alternatives
- `@doc` cannot be used on some words, such as `let` or `for`

The method for displaying the documentation need not be the same, for example it can be an Emacs function bound to shortcut such as `C-c C-h`, looking up the word at point or querying the user.

The standard shell also has a _shell mode_, which can be entered by pressing `;` at an empty prompt. But running shell commands is already present in Emacs, so it is low priority.

See also [the Julia manual on help mode](http://docs.julialang.org/en/release-0.4/manual/interacting-with-julia/#help-mode).

**Comment by dennisog** [comment]
I agree. I recently ran into the same problem. Before v0.4, I was able to circumvent that issue by just using `help()` from the REPL, but that feature was removed in v0.4. It seems like most of the docs/help code happens [here](https://github.com/JuliaLang/julia/tree/master/base/docs), and I'll try to look into it soon. Any suggestions?

EDIT: I'm probably going to build something that does [this](https://github.com/JuliaLang/julia/blob/master/base/REPL.jl#L716)

**Comment by salvipeter** [comment]
Thanks for the quick reply!

Based on the code you've mentioned, maybe you can do the following:
1. Define a help function as soon as the shell starts:
   
   ```
   julia_shell_help(ex) = eval(:(Base.Docs.@repl $(symbol(ex))))
   ```
   
   This function accepts either strings (e.g. "for") or symbols (e.g. :let) as arguments.
2. Substitute (internally) `?foo` with `julia_shell_help("foo")`. History should show the original `?foo` version.

**Comment by microamp** [comment]
I'd love to see this feature implemented too. Just started learning Julia. :grinning:
